util/patricia_trie/src/lib.rs

```diff
@@ -245,40 +251,41 @@ macro_rules! wrapper {
 	}
 }
 
-impl<'db> Trie for TrieKinds<'db> {
-	fn root(&self) -> &H256 {
+impl<'db, H: Hasher, C: NodeCodec<H>> Trie for TrieKinds<'db, H, C> where H::Out: rlp::Decodable + rlp::Encodable {
```

Comment:
why are these `rlp` bounds necessary?

Reply:
Largely they are not needed (put there before `NodeCodec` was added) and I'll remove them.

However there's the problem of `into_rlp()` and `commit_node()` methods that still make use of `RlpStream` and they introduce the `Encodable` bound (which then bubble up).

Partly fixed in #5db5de901